### ISSUE TITLE
ModelsCommand->checkForCastableCasts does not support AsArrayObject

### ISSUE THREAD

**Issue by CrixuAMG** [barryvdh/laravel-ide-helper#1305]
### Versions:
- ide-helper Version: 2.12.0
- Laravel Version: 8.80.0
- PHP Version: 7.4

### Description and steps to reproduce:

When setting the cast of a JSON object to `\Illuminate\Database\Eloquent\Casts\AsArrayObject`, ide-helper:models fails to get the return type of that cast. At line 1293 of the ModelsCommand class both methods return `null`, even though the AsArrayObject class contains the following `@return` tag in the docblock.

```
    /*
     * @return object|string
     */
```

The error:

```
> php artisan ide-helper:models --nowrite
Exception: Return value of Barryvdh\LaravelIdeHelper\Console\ModelsCommand::checkForCastableCasts() must be of the type string, null returned
```

**Comment by mfn** [comment]
@binotaliu is this related to https://github.com/barryvdh/laravel-ide-helper/pull/1287 ? 🤔

**Comment by CrixuAMG** [comment]
Did a little bit more looking into how the return type is detected, it seems logical that the command is failing since the get function in the AsArrayObject class has no doc block and no return type. A possible fix could be setting a default `mixed` value for when no return value could be detected, another would be opening a pull request to `illuminate/database` containing doc blocks/return types, what would the preferred option be? Or is there a better/easier fix?


Here's the full class for reference:

```
<?php

namespace Illuminate\Database\Eloquent\Casts;

use Illuminate\Contracts\Database\Eloquent\Castable;
use Illuminate\Contracts\Database\Eloquent\CastsAttributes;

class AsArrayObject implements Castable
{
    /**
     * Get the caster class to use when casting from / to this cast target.
     *
     * @param  array  $arguments
     * @return object|string
     */
    public static function castUsing(array $arguments)
    {
        return new class implements CastsAttributes
        {
            public function get($model, $key, $value, $attributes)
            {
                return isset($attributes[$key]) ? new ArrayObject(json_decode($attributes[$key], true)) : null;
            }

            public function set($model, $key, $value, $attributes)
            {
                return [$key => json_encode($value)];
            }

            public function serialize($model, string $key, $value, array $attributes)
            {
                return $value->getArrayCopy();
            }
        };
    }
}

```

**Comment by binotaliu** [comment]
@mfn yes. return type should be `?string`. I'll open a new PR to fix this soon.

**Comment by vadimwe** [comment]
Same problem:

`Exception: Barryvdh\LaravelIdeHelper\Console\ModelsCommand::checkForCastableCasts(): Return value must be of type string, null returned
Could not analyze class App\Models\Projects\Project.`

Cast:
`public function get($model, string $key, $value, array $attributes)
    {
        return collect(json_decode($value, true))->map(function ($serviceItem) {
            return new InvoiceServiceItem($serviceItem['title'], $serviceItem['qty'], $serviceItem['itemPrice']);
        });
    }

public function set($model, string $key, $value, array $attributes)
    {
        return json_encode($value);
    }
`

**Comment by mfn** [comment]
@vadimwe doe the check in https://github.com/barryvdh/laravel-ide-helper/pull/1306 work for you?

**Comment by mfn** [comment]
Bug fix 🤞🏼 has just been released as https://github.com/barryvdh/laravel-ide-helper/releases/tag/v2.12.1

**Comment by hotmeteor** [comment]
Wow, came here looking for this. Fixed 28 minutes ago 😂 Thanks all!

**Comment by CrixuAMG** [comment]
Thanks for the quick fix! 🙏

**Comment by vadimwe** [comment]
> @vadimwe doe the change in #1306 work for you?

Yes! thank you!
